08-rails/debugging-rails-vscode.md

```diff
@@ -0,0 +1,118 @@
+# Debugging Rails with VS Code
+## Learning Goals
+- Practice setting breakpoints and watch values.
+- Additional experience tracking the flow of logic in a Rails application.
+
+
+## Adding New tools
+VS Code and Ruby are still immature in their compatibility, but with a few extensions we can leverage the debugger that comes with VS Code to make our lives a lot easier.
+
+To start off, we should make sure that we have the [ruby](https://marketplace.visualstudio.com/items?itemName=rebornix.Ruby) VS Code extension installed. Use `⇧⌘X` to open the extension window and check to see that it is currently installed.
+
+Additionally, we need to modify the `Gemfile`:
+
+```ruby
+#Gemfile
+group :development, :test do
+  gem 'ruby-debug-ide'
+  gem 'debase'
+end
+```
+
+The [debase](https://rubygems.org/gems/debase) gem is an implementation of the standard Ruby debugger that works with Ruby 2.0, and [ruby-debug-ide](https://rubygems.org/gems/ruby-debug-ide) gem he helps `debase` communicate with VS Code so that we can see and set breakpoints in the IDE itself.
+
+__NOTE: These gems should ALWAYS be part of development and test ONLY.__
+
+Once they are included in your `Gemfile`, run the following commands:
+
+```ruby
+  bundle install 
+  #installs the new gems we just added to the Gemfile
+
+  $: which bundle
```

Comment:
Kaida mentioned this before, but it may make more sense to word this command as `$ which bundle` etc